feat(apple): Update carthage install process (#7496)
getsentry/sentry-docs: src/platforms/apple/common/install/carthage.mdx

github "getsentry/sentry-cocoa" "{{@inject packages.version('sentry.cocoa') }}"
```

Run `carthage update` to download the framework and drag the built _Sentry.xcframework_ into your Xcode project. We also provide a pre-built version for every release, which can be downloaded at [releases on GitHub](https://github.com/getsentry/sentry-cocoa/releases).

Our Carthage setup generates 3 different frameworks: - Sentry - SentryPrivate - SentrySwiftUI

> You only need to add _SentryPrivate_ to your project if you changed carthage build process to generate static frameworks. Otherwise, you can ignore it.
> You add _SentrySwiftUI_ to your project if you want to measure performance of your SwiftUI views.

Reviewer: ```suggestion
_Sentry_ is required for all projects.  _SentrySwiftUI_ is required to measure the performance of your SwiftUI views. _SentryPrivate_ is only needed if you changed the carthage build process to generate static frameworks.
```
